bitcoin/src/blockdata/locktime/absolute.rs

```diff
@@ -329,7 +330,7 @@ impl FromHexStr for LockTime {
     type Error = ParseIntError;
 
     #[inline]
-    fn from_hex_str_no_prefix<S: AsRef<str> + Into<String>>(s: S) -> Result<Self, Self::Error> {
```

Comment:
If we can avoid `Into<InputString>` I'd like to. It would require the implementor of whatever string type users are using to be aware of `InputString`. (Though maybe we only care about `String` and `&str`.) For example we don't have an `Into<InputString>` impl for `CString`. Not sure if one is possible, but if so, and if a user tried to use this, they'd be SOL.

Reply:
This is private function, so it doesn't really matter. IIRC we can't do blanket impls anyway.

`CString` doesn't guarantee UTF-8 so we can't use it. (Did I say that Rust doesn't have enough string types already?) And as I said elsewhere people using funny types can just deref them. It's unlikely they are zero-cost-convertible to `String` anyway.

Side note: I was thinking of making `InputString(InputStringRepr)` with `enum InputStringRepr { String(String), Arc(Arc) }` but that seems too complex and won't be used 99.99% of the time. (There are `unsafe` hacks to make it almost zero-cost but I'm not eager to use them either.)

Reply:
Oh, yeah, for private functions `Into<InputString>` is fine.